hudi-common/src/main/java/org/apache/hudi/common/config/ConfigGroups.java

```diff
@@ -40,7 +40,8 @@ public enum Names {
     RECORD_PAYLOAD("Record Payload Config"),
     KAFKA_CONNECT("Kafka Connect Configs"),
     AWS("Amazon Web Services Configs"),
-    HUDI_STREAMER("Hudi Streamer Configs");
+    HUDI_STREAMER("Hudi Streamer Configs"),
+    INDEXING("Indexing Configs");
```

Comment:
There is `INDEX` sub-group name below.  Should that be consolidated?
```
 INDEX(
        "Index Configs",
        "Configurations that control indexing behavior, "
            + "which tags incoming records as either inserts or updates to older records."),
```

Reply:
Yeah I'm slightly biased towards INDEXING/INDEXER instead of INDEX. Let me take that it in a seprate PR. It will touch more files.

Reply:
In that case, let's remove the subgroup of `INDEX` or rename it sth different in the follow-up.